src/modm/driver/pressure/ms5837_impl.hpp

```diff
@@ -0,0 +1,154 @@
+// coding: utf-8
+// ----------------------------------------------------------------------------
+/*
+ * Copyright (c) 2023, Vivien Henry
+ * Based on the ms5837, courtesy of Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_MS5837_HPP
+#   error "Don't include this file directly, use 'ms5837.hpp' instead!"
+#endif
+
+#include <modm/math/utils/endianness.hpp>
+
+namespace modm
+{
+
+template < typename I2cMaster >
+Ms5837<I2cMaster>::Ms5837(DataBase &data, uint8_t address) : I2cDevice<I2cMaster, 2>(address), data(data)
+{
+}
+
+// -----------------------------------------------------------------------------
+
+template < typename I2cMaster >
+modm::ResumableResult<uint16_t>
+Ms5837<I2cMaster>::initialize()
+{
+    RF_BEGIN();
+
+    // Reset sensor once after power-on to load calibration PROM into the internal registers
+    buffer[0] = i(Command::Reset);
+    this->transaction.configureWrite(buffer, 1);
+    if (RF_CALL( this->runTransaction() ))
+    {
+        MODM_LOG_DEBUG.printf("MS5837 Reset Success\n");
+    }
+    else {
+        MODM_LOG_DEBUG.printf("MS5837 Reset Failed\n");
+        RF_RETURN(false);
```

Comment:
```suggestion
        RF_RETURN(0);
```
instead of `false`, since the return type is `uint16_t`.
Not sure how likely it is to get a valid `0` as the results of the `data.prom.calculateCrc()` function though. It's probably fine.